### ISSUE TITLE
cannot  intall lilypond from lyp

### ISSUE THREAD

**Issue by Nyavlys** [noteflakes/lyp#55]
Hi,

Thanks for lyp, which seems to have very interesting features.

I installed it, but could not install lilypond:

```
$ lyp version
Lyp 1.3.5

$ lyp install lilypond
undefined method `[]' for nil:NilClass

$lyp install -V lilypond
undefined method `[]' for nil:NilClass
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/lilypond.rb:425:in `latest_stable_version'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/lilypond.rb:468:in `detect_version_from_specifier'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/lilypond.rb:448:in `install'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/cli.rb:211:in `block in install'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/cli.rb:205:in `each'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/cli.rb:205:in `install'
/var/lib/gems/2.3.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
/var/lib/gems/2.3.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
/var/lib/gems/2.3.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
/var/lib/gems/2.3.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/cli.rb:428:in `run'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/bin/lyp:9:in `<top (required)>'
/usr/local/bin/lyp:22:in `load'
/usr/local/bin/lyp:22:in `<main>'
```

In your answer to another issue (#32), you mentionned rugged. I have it installed:
```
$ sudo gem install rugged
Building native extensions.  This could take a while...
Successfully installed rugged-0.26.0
Parsing documentation for rugged-0.26.0
Done installing documentation for rugged after 1 seconds
1 gem installed

$ gem list rugged

*** LOCAL GEMS ***

rugged (0.26.0, 0.23.0)
```

So what am I missing?

Thanks.

**Comment by ciconia** [comment]
Hmm it seems like lyp has trouble finding versions of Lilypond to install. For the moment I haven't been able to reproduce the problem. Can you run `lyp search -V lilypond`?

**Comment by Nyavlys** [comment]
```
$ lyp search -V lilypond

No available versions of lilypond@ found
```
It seems also to have problems finding other packages:
```
$ lyp install lilyjazz
Cloning https://github.com/lyp-packages/lilyjazz.git...
Could not clone repository (please check that the package URL is correct.)
$ lyp install -V lilyjazz
Cloning https://github.com/lyp-packages/lilyjazz.git...
there is no TLS stream available
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/package.rb:240:in `clone_at'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/package.rb:240:in `package_repository'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/package.rb:142:in `install_from_repository'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/package.rb:70:in `install'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/cli.rb:214:in `block in install'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/cli.rb:205:in `each'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/cli.rb:205:in `install'
/var/lib/gems/2.3.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
/var/lib/gems/2.3.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
/var/lib/gems/2.3.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
/var/lib/gems/2.3.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/lib/lyp/cli.rb:428:in `run'
/var/lib/gems/2.3.0/gems/lyp-1.3.5/bin/lyp:9:in `<top (required)>'
/usr/local/bin/lyp:22:in `load'
/usr/local/bin/lyp:22:in `<main>'

```

**Comment by ciconia** [comment]
Can you do the following in a temporary directory?

    git clone https://github.com/lyp-packages/lilyjazz.git

**Comment by Nyavlys** [comment]
Yes, it works:
```
$ git clone https://github.com/lyp-packages/lilyjazz.git
Cloning into 'lilyjazz'...
remote: Counting objects: 34, done.
remote: Total 34 (delta 0), reused 0 (delta 0), pack-reused 34
Unpacking objects: 100% (34/34), done.
```

**Comment by ciconia** [comment]
Fixed in commit 1223803945c9495ff3108160b9f86ec08e72288e.

**Comment by baloe** [comment]
same here. fresh install on a fresh ubuntu 18.04 system.
After:

```
sudo apt-get install -y ruby
sudo apt-get install -y ruby-dev
sudo apt-get install -y cmake
sudo apt-get install -y pkg-config
sudo gem install lyp
```

```
bastian@laptwo ~> lyp install lilypond
undefined method `[]' for nil:NilClass
```
and
```
bastian@laptwo ~> lyp search -V lilypond

No available versions of lilypond@ found
```
